modelon/impact/client/entities/case.py

```diff
@@ -455,16 +536,63 @@ def get_artifact(self, artifact_id: str) -> Tuple[bytes, str]:
 
         Example::
 
-            result, file_name = case.get_artifact("ABCD")
-            with open(file_name, "wb") as f:
-                f.write(result)
+            custom_artifact = case.get_artifact(artifact_id)
         """
         assert_successful_operation(self.is_successful(), self._case_id)
-        result, file_name = self._sal.experiment.case_artifact_get(
-            self._workspace_id, self._exp_id, self._case_id, artifact_id
+        if not download_as:
+            resp = self._sal.experiment.case_artifacts_meta_get(
+                self._workspace_id, self._exp_id, self._case_id
+            )
+            meta: List[Dict[str, str]] = [
```

Comment:
Construct:  `x = [x for x in ... ]` implies different data type for `x` (collection vs single item) which is a bit confusing. I suggest in this case:

`meta = next((meta for meta in resp["data"]["items"] if meta["id"] == artifact_id), None)`